src/components/GradeForm.js

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from 'react';
+import Grade from './Grade';
+
+class GradeForm extends Component {
+
+    state = {
+        label: 'JavaScript Quiz',
+        score: '3',
+        grades: [
+            { label: 'Important Test', score: 20},
+            { label: 'Nuclear Energy Exam', score: 1}
+        ]
+
+    }
+    handleChangeLabel = e => {
+        this.setState({
+            label: e.target.value
+        });
+    }
+    handleChangeScore = e => {
+        this.setState({
+            score: e.target.value,
+
+        });
+    }
+    handleSubmit = (e) => {
+        e.preventDefault();
+        let pass
+
+        if (this.state.score >= 12) {
+            pass = true
+        }
+    
+        const newItem = {
+          label: this.state.label,
+          score: this.state.score,
+          passing: pass
+          
+        };
+    
+        // return an object that gives new value (to set state)
+        this.setState(prev => {
+          const newList = prev.grades.slice(0);
```

Comment:
This works perfectly fine, so if you're happy with it there's no need to change it. 

I have transitioned to using spread operators to do this work because creating a copy of the array/pushing a new element into it seemed like I was "juggling" variables. Here's how it could work with spread:

```
this.setState(currentState => {
    return {
        ...currentState, // first, "spread" the currentState into the new object literal. If we did nothing else, this would return an exact copy of `currentState` without the need to slice/copy
        grades: [ // lets use the same spread principle _inside_ the object to create a new `grades` array.
            ...currentState.grades, // pull in the current grades array and spread it in place
            { // merge in a new object to the end of the array
                label: this.state.label,
                score: this.state.score,
                passing: pass
            }
        ]
    }
});
```